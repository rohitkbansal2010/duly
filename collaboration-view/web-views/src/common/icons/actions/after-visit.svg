<svg width="25" height="27" viewBox="0 0 25 27" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M13.0527 2H5C3.34315 2 2 3.34315 2 5V22C2 23.6569 3.34314 25 5 25H13.0527" stroke="white" stroke-width="2.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M10.1055 13.871H23.0002M23.0002 13.871L17.1055 7.93555M23.0002 13.871L17.4739 19.8065" stroke="white" stroke-width="2.5" stroke-linecap="round"/>
</svg>
