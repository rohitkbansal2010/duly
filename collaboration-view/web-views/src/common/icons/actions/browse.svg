<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M20.4678 8.96715C20.4678 8.96715 20.4678 8.96715 20.4678 8.92202C19.8324 7.18556 18.6795 5.68634 17.165 4.62717C15.6505 3.568 13.8475 3 12 3C10.1525 3 8.3495 3.568 6.83501 4.62717C5.32052 5.68634 4.1676 7.18556 3.5322 8.92202C3.5322 8.92202 3.5322 8.92202 3.5322 8.96715C2.8226 10.9267 2.8226 13.0733 3.5322 15.0328C3.5322 15.0328 3.5322 15.0328 3.5322 15.078C4.1676 16.8144 5.32052 18.3137 6.83501 19.3728C8.3495 20.432 10.1525 21 12 21C13.8475 21 15.6505 20.432 17.165 19.3728C18.6795 18.3137 19.8324 16.8144 20.4678 15.078C20.4678 15.078 20.4678 15.078 20.4678 15.0328C21.1774 13.0733 21.1774 10.9267 20.4678 8.96715ZM5.00211 13.8053C4.6894 12.6221 4.6894 11.3779 5.00211 10.1947H6.67944C6.53518 11.3939 6.53518 12.6061 6.67944 13.8053H5.00211ZM5.74158 15.6105H7.00409C7.21575 16.4155 7.5183 17.1938 7.90588 17.9303C7.02149 17.3273 6.2822 16.5349 5.74158 15.6105ZM7.00409 8.38947H5.74158C6.27442 7.46796 7.00426 6.67579 7.87882 6.0697C7.50057 6.80737 7.20712 7.5856 7.00409 8.38947ZM11.0802 18.9503C9.97257 18.1367 9.19467 16.9512 8.88883 15.6105H11.0802V18.9503ZM11.0802 13.8053H8.50106C8.33277 12.6076 8.33277 11.3924 8.50106 10.1947H11.0802V13.8053ZM11.0802 8.38947H8.88883C9.19467 7.04885 9.97257 5.86329 11.0802 5.04973V8.38947ZM18.2223 8.38947H16.9598C16.7482 7.58446 16.4456 6.80619 16.0581 6.0697C16.9424 6.67272 17.6817 7.46512 18.2223 8.38947ZM12.8838 5.04973C13.9914 5.86329 14.7693 7.04885 15.0751 8.38947H12.8838V5.04973ZM12.8838 18.9503V15.6105H15.0751C14.7693 16.9512 13.9914 18.1367 12.8838 18.9503ZM15.4629 13.8053H12.8838V10.1947H15.4629C15.6312 11.3924 15.6312 12.6076 15.4629 13.8053ZM16.0851 17.9303C16.4727 17.1938 16.7752 16.4155 16.9869 15.6105H18.2494C17.7088 16.5349 16.9695 17.3273 16.0851 17.9303ZM18.9618 13.8053H17.2845C17.3578 13.2063 17.394 12.6034 17.3927 12C17.3937 11.3966 17.3576 10.7937 17.2845 10.1947H18.9618C19.2745 11.3779 19.2745 12.6221 18.9618 13.8053Z" fill="#002855"/>
</svg>
