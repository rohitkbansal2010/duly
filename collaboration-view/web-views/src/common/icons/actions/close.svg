<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M13.409 11.9991L19.7044 5.71303C19.8926 5.52485 19.9983 5.26961 19.9983 5.00347C19.9983 4.73733 19.8926 4.4821 19.7044 4.29391C19.5162 4.10572 19.261 4 18.9949 4C18.7288 4 18.4736 4.10572 18.2854 4.29391L12 10.59L5.71456 4.29391C5.52639 4.10572 5.27118 4 5.00507 4C4.73896 4 4.48375 4.10572 4.29558 4.29391C4.10742 4.4821 4.00171 4.73733 4.00171 5.00347C4.00171 5.26961 4.10742 5.52485 4.29558 5.71303L10.591 11.9991L4.29558 18.2853C4.20192 18.3782 4.12758 18.4887 4.07685 18.6105C4.02612 18.7323 4 18.8629 4 18.9948C4 19.1268 4.02612 19.2574 4.07685 19.3792C4.12758 19.5009 4.20192 19.6115 4.29558 19.7044C4.38848 19.7981 4.499 19.8724 4.62077 19.9231C4.74254 19.9739 4.87315 20 5.00507 20C5.13699 20 5.2676 19.9739 5.38937 19.9231C5.51114 19.8724 5.62166 19.7981 5.71456 19.7044L12 13.4083L18.2854 19.7044C18.3783 19.7981 18.4889 19.8724 18.6106 19.9231C18.7324 19.9739 18.863 20 18.9949 20C19.1268 20 19.2575 19.9739 19.3792 19.9231C19.501 19.8724 19.6115 19.7981 19.7044 19.7044C19.7981 19.6115 19.8724 19.5009 19.9232 19.3792C19.9739 19.2574 20 19.1268 20 18.9948C20 18.8629 19.9739 18.7323 19.9232 18.6105C19.8724 18.4887 19.7981 18.3782 19.7044 18.2853L13.409 11.9991Z" fill="#002855"/>
</svg>
