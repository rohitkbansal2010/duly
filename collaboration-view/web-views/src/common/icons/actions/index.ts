import addIcon from './add.svg';
import AfterVisitLinkIcon from './after-visit.svg';
import attachIcon from './attach.svg';
import browseIcon from './browse.svg';
import closeIcon from './close.svg';
import cropIcon from './crop.svg';
import downloadIcon from './download.svg';
import editIcon from './edit.svg';
import expandSolidIcon from './expand-solid.svg';
import expandIcon from './expand.svg';
import ExportPdf from './export-pdf.svg';
import favoriteIcon from './favorite.svg';
import filterOutlinedIcon from './filter-outlined.svg';
import filterSolidIcon from './filter-solid.svg';
import FontIconBlue from './font-icon-blue.svg';
import FontIconSkyblue from './font-icon-skyblue.svg';
import lineToolIcon from './line-tool.svg';
import Line from './Line.svg';
import messageIcon from './message.svg';
import moreIcon from './more.svg';
import noteIcon from './note.svg';
import printIcon from './print.svg';
import reflectIcon from './reflect.svg';
import reviewVitalsIcon from './review-vitals.svg';
import searchIcon from './search.svg';
import sendIcon from './send.svg';
import shapeToolIcon from './shape-tool.svg';
import textToolIcon from './text-tool.svg';
import updateIcon from './update.svg';
import uploadIcon from './upload.svg';
import viewPdfIcon from './view-pdf.svg';
import voiceRecorderIcon from './voice-recorder.svg';

export {
  reviewVitalsIcon,
  viewPdfIcon,
  voiceRecorderIcon,
  textToolIcon,
  updateIcon,
  uploadIcon,
  shapeToolIcon,
  searchIcon,
  sendIcon,
  printIcon,
  reflectIcon,
  addIcon,
  noteIcon,
  lineToolIcon,
  messageIcon,
  moreIcon,
  filterSolidIcon,
  attachIcon,
  filterOutlinedIcon,
  browseIcon,
  closeIcon,
  cropIcon,
  downloadIcon,
  favoriteIcon,
  editIcon,
  expandIcon,
  expandSolidIcon,
  FontIconSkyblue,
  FontIconBlue,
  AfterVisitLinkIcon,
  ExportPdf,
  Line
};
