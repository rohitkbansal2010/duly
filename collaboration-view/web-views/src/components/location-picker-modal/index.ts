export * from './location-picker-modal';
