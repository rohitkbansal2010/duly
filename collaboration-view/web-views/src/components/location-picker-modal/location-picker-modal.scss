@import "~@styles/colors.scss";
@import "~@styles/typography.scss";
@import "~@styles/main.scss";

.customWidthModal {
  width: 37.5rem !important;
  max-width: 37.5rem !important;

  & > div:first-of-type {
    height: 29.5rem !important;
  }
}

.customBackDropModal {
  background-color: $default-text !important;
  opacity: 0.15 !important;
}

.locationPickerModal {
  &Header {
    @include nexa-heavy-highlight;

    display: flex;
    flex-flow: column nowrap;
    margin: 0 auto;
    padding: 2rem 0 !important;
    font-size: 2.5rem !important;
    border-bottom: none !important;

    &Icon {
      display: flex;
      justify-content: center;
      width: 10rem;
      height: 10rem;
      background-color: rgba($dupage-magenta, 0.3);
      border: none;
      border-radius: 100%;

      img {
        display: flex;
        width: 4.6875rem;
        color: $dupage-magenta;
      }
    }

    &Title {
      padding-top: 2.5rem;
      color: $black;
      font-size: 2.5rem;
    }
  }

  &Body {
    padding-top: 0 !important;
  }

  &Dropdown {
    &Item {
      width: 100% !important;
      height: 100% !important;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
      border: unset !important;
      box-shadow: unset !important;
    }

    &Button {
      display: flex;
      justify-content: center;

      button {
        @include inter-text-default;
        position: relative;
        display: block;

        width: 26.5rem;
        height: 3rem;
        padding-right: 3rem;

        overflow-x: hidden;
        color: $default-text;
        white-space: nowrap;
        text-align: left;
        text-overflow: ellipsis;

        &:after {
          position: absolute;
          top: 50%;
          right: 1.95rem;
          transform: translateY(-50%);
        }

        &:focus {
          box-shadow: unset;
        }
      }

      & > div:first-of-type {
        @include ios-scrollbar;
        box-sizing: border-box;
        max-height: 6.5rem;
        padding: 0;

        overflow-x: hidden;
        overflow-y: auto !important;
        white-space: nowrap;
        text-overflow: ellipsis;
        border: 0.0625rem solid $light-gray;

        border-radius: 0.25rem;
        box-shadow: 0 0 0.0625rem convertToRGBA($dark-gray, 0.06),
          0 0.625rem 0.9375rem -0.1875rem convertToRGBA($dark-gray, 0.1),
          0 0.25rem 0.375rem -0.125rem convertToRGBA($dark-gray, 0.05);
      }

      a {
        @include inter-text-default;

        width: 26.5rem;
        color: $default-text;

        &:hover {
          background-color: $dark-blue-tint;
        }
      }
    }
  }
}
