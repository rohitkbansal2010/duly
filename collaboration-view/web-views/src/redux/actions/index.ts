export * from './ui';
export * from './locale';
export * from './types';
export * from './user';
export * from './app-state';
export * from './todays-appointments';
export * from './current-appointment';
export * from './overview-widgets';
export * from './site';
