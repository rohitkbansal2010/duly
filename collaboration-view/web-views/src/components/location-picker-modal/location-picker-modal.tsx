import { useState, useCallback } from 'react';
import { Dropdown, DropdownButton, Modal } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import { SITE_ID } from '@constants';
import { locationPickerIcon } from '@icons';
import { hideLocationPickerModal, setCurrentSite } from '@redux/actions';
import { RootState } from '@redux/reducers';
import { Site } from '@types';
import { formatAddress, setLocalStorageItem } from '@utils';

import styles from './location-picker-modal.scss';

export const LocationPickerModal = () => {
  const { isLocationPickerModalShown, sites, currentSite } = useSelector(({ SITE }: RootState) =>
    SITE);

  const dispatch = useDispatch();

  const [ location, setLocation ] = useState<Site | null>(null);
  const site = currentSite || location;

  const handleSelect = useCallback((location: Site) => {
    setLocation(location);
    setLocalStorageItem(SITE_ID, location.id);
    dispatch(setCurrentSite(location));
    dispatch(hideLocationPickerModal());
  }, [ dispatch, setLocation ]);

  const handleClose = useCallback(() => {
    if (site) {
      dispatch(hideLocationPickerModal());
    }
  }, [ site, dispatch ]);

  const dropdownTitle = site?.address ? formatAddress(site.address) : 'Selection';

  return (
    <Modal
      animation={false}
      show={isLocationPickerModalShown}
      className={styles.locationPickerModal}
      size="lg"
      centered
      dialogClassName={styles.customWidthModal}
      backdropClassName={styles.customBackDropModal}
      onHide={handleClose}
    >
      <Modal.Header className={styles.locationPickerModalHeader}>
        <div className={styles.locationPickerModalHeaderIcon}>
          <img
            src={locationPickerIcon}
            alt="location picker icon"
          />
        </div>
        <h1
          className={styles.locationPickerModalHeaderTitle}
        >
          Please select your location
        </h1>
      </Modal.Header>
      <Modal.Body className={styles.locationPickerModalBody}>
        <DropdownButton
          variant="outline-secondary"
          title={dropdownTitle}
          className={styles.locationPickerModalDropdownButton}
        >
          {sites?.map((item: Site) =>
            (
              <Dropdown.Item key={item?.id}>
                <div
                  role="none"
                  data-testid="location-picker-modal__dropdown__dropdown-item"
                  className={styles.locationPickerModalDropdownItem}
                  onClick={() => { handleSelect(item); }}
                >
                  {!!item.address && formatAddress(item?.address)}
                </div>
              </Dropdown.Item>
            ))}
        </DropdownButton>
      </Modal.Body>
    </Modal>
  );
};
