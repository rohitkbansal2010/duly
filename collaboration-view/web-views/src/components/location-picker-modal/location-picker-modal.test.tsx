import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Modal } from 'react-bootstrap';
import { act } from 'react-dom/test-utils';
import { useSelector } from 'react-redux';

import { LocationPickerModal } from '@components/location-picker-modal';
import { SITE_ID } from '@constants';
import { hideLocationPickerModal, setCurrentSite } from '@redux/actions';
import { formatAddress, setLocalStorageItem } from '@utils';


const mockDispatch = jest.fn();
jest.mock('react-redux', () =>
  ({
    ...jest.requireActual('react-redux'),
    useDispatch: () =>
      mockDispatch,
    useSelector: jest.fn(),
  }));

jest.mock('../../common/utils/local-storage');

const mockSites = [
  {
    id: '1',
    address: {
      city: 'test city 1',
      line: 'test line 1',
      postalCode: 'test postalCode 1',
      state: 'test state 1',
    },
  },
  {
    id: '2',
    address: {
      city: 'test city 2',
      line: 'test line 2',
      postalCode: 'test postalCode 2',
      state: 'test state 2',
    },
  },
  {
    id: '3',
    address: {
      city: 'test city 3',
      line: 'test line 3',
      postalCode: 'test postalCode 3',
      state: 'test state 3',
    },
  },
];

describe('LocationPickerModal', () => {
  let Component: ReturnType<typeof mount>;
  const mockState = { isLocationPickerModalShown: true, sites: mockSites, currentSite: null };

  beforeEach(() => {
    (useSelector as jest.Mock).mockImplementation((selector: (state: unknown) => unknown) =>
      selector({ SITE: mockState }));
    Component = mount(
      <LocationPickerModal />
    );
  });

  it ('should render correctly', () => {
    expect(toJson(Component)).toMatchSnapshot();
  });

  it('should have default title without location', () => {
    const dropdownButton = Component.find('.locationPickerModalDropdownButton').find('button');

    expect(dropdownButton.text()).toBe('Selection');
  });

  it('should render list of sites', async () => {
    const dropdownButton = Component.find('.locationPickerModalDropdownButton').find('button');

    dropdownButton.simulate('click');

    const dropdownItems = Component.find('.locationPickerModalDropdownButton').find({ 'data-testid': 'location-picker-modal__dropdown__dropdown-item' });

    expect(dropdownItems.length).toBe(mockSites.length);

    await act(async () =>
      undefined);
  });

  it('should not dispatch close modal action if no site was selected', () => {
    const ModalWrapper = Component.find(Modal);

    ModalWrapper.props().onHide();
    expect(mockDispatch).not.toHaveBeenCalled();
  });

  it('should render correct title if site is present', () => {
    (useSelector as jest.Mock).mockImplementation((selector: (state: unknown) => unknown) =>
      selector({ SITE: { ...mockState, currentSite: mockSites[0] } }));

    const wrapper = mount(<LocationPickerModal />);

    const dropdownButton = wrapper.find('.locationPickerModalDropdownButton').find('button');

    expect(dropdownButton.text()).toBe(formatAddress(mockSites[0].address));
  });

  it('should dispatch correct action if site is present', () => {
    (useSelector as jest.Mock).mockImplementation((selector: (state: unknown) => unknown) =>
      selector({ SITE: { ...mockState, currentSite: mockSites[0] } }));

    const wrapper = mount(<LocationPickerModal />);

    const ModalWrapper = wrapper.find(Modal);

    ModalWrapper.props().onHide();
    expect(mockDispatch).toHaveBeenCalledWith(hideLocationPickerModal());
  });
});


describe('should handle click', () => {
  const mockState = { isLocationPickerModalShown: true, sites: mockSites, currentSite: null };

  beforeEach(() => {
    (useSelector as jest.Mock).mockImplementation((selector: (state: unknown) => unknown) =>
      selector({ SITE: mockState }));
  });

  test.each(mockSites.map((site, index) =>
    ({ index, site: { ...site } })))('should handle click for test site $site.id', async ({ index, site }) => {
    const formattedAddress = formatAddress(site.address);

    const Component = mount(<LocationPickerModal />);

    const dropdownButton = Component.find('.locationPickerModalDropdownButton').find('button');

    dropdownButton.simulate('click');

    const dropdownItems = Component.find('.locationPickerModalDropdownButton').find({ 'data-testid': 'location-picker-modal__dropdown__dropdown-item' });
    expect(dropdownItems.at(index).text()).toBe(formattedAddress);

    dropdownItems.at(index).simulate('click');

    expect(setLocalStorageItem).toHaveBeenCalledWith(SITE_ID, site.id);
    expect(mockDispatch).toHaveBeenCalledWith(setCurrentSite(site));
    expect(mockDispatch).toHaveBeenCalledWith(hideLocationPickerModal());

    Component.update();

    const updatedDropdownButton = Component.find('.locationPickerModalDropdownButton').find('button');
    expect(updatedDropdownButton.text()).toBe(formattedAddress);

    await act(async () =>
      undefined);
  }, 0);
});
