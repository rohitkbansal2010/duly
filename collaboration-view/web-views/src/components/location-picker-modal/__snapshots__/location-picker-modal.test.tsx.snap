// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationPickerModal should render correctly 1`] = `
<LocationPickerModal>
  <Modal
    animation={false}
    autoFocus={true}
    backdrop={true}
    backdropClassName="customBackDropModal"
    centered={true}
    className="locationPickerModal"
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    dialogClassName="customWidthModal"
    enforceFocus={true}
    keyboard={true}
    onHide={[Function]}
    restoreFocus={true}
    show={true}
    size="lg"
  >
    <Modal
      autoFocus={true}
      backdrop={true}
      enforceFocus={true}
      keyboard={true}
      manager={
        BootstrapModalManager {
          "handleContainerOverflow": true,
          "isRTL": false,
          "modals": Array [
            Object {
              "add": [Function],
              "backdrop": <div
                class="modal-backdrop customBackDropModal show"
              />,
              "dialog": <div
                aria-modal="true"
                class="locationPickerModal modal"
                role="dialog"
                style="display: block;"
                tabindex="-1"
              >
                <div
                  class="modal-dialog customWidthModal modal-lg modal-dialog-centered"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      class="locationPickerModalHeader modal-header"
                    >
                      <div
                        class="locationPickerModalHeaderIcon"
                      >
                        <img
                          alt="location picker icon"
                          src="location-picker.svg"
                        />
                      </div>
                      <h1
                        class="locationPickerModalHeaderTitle"
                      >
                        Please select your location
                      </h1>
                    </div>
                    <div
                      class="locationPickerModalBody modal-body"
                    >
                      <div
                        class="locationPickerModalDropdownButton dropdown"
                      >
                        <button
                          aria-expanded="false"
                          class="dropdown-toggle btn btn-outline-secondary"
                          type="button"
                        >
                          Selection
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>,
              "isTopModal": [Function],
              "remove": [Function],
              "setBackdropRef": [Function],
              "setDialogRef": [Function],
            },
          ],
          "state": Object {
            "scrollBarWidth": 1024,
            "style": Object {
              "overflow": "",
              "paddingRight": "",
            },
          },
        }
      }
      onEnter={[Function]}
      onEntering={[Function]}
      onEscapeKeyDown={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onHide={[Function]}
      renderBackdrop={[Function]}
      renderDialog={[Function]}
      restoreFocus={true}
      show={true}
    >
      <Portal
        containerInfo={
          <body
            class="modal-open"
            data-rr-ui-modal-open=""
            style="padding-right: 1024px;"
          >
            <div
              class="modal-backdrop customBackDropModal show"
            />
            <div
              aria-modal="true"
              class="locationPickerModal modal"
              role="dialog"
              style="display: block;"
              tabindex="-1"
            >
              <div
                class="modal-dialog customWidthModal modal-lg modal-dialog-centered"
              >
                <div
                  class="modal-content"
                >
                  <div
                    class="locationPickerModalHeader modal-header"
                  >
                    <div
                      class="locationPickerModalHeaderIcon"
                    >
                      <img
                        alt="location picker icon"
                        src="location-picker.svg"
                      />
                    </div>
                    <h1
                      class="locationPickerModalHeaderTitle"
                    >
                      Please select your location
                    </h1>
                  </div>
                  <div
                    class="locationPickerModalBody modal-body"
                  >
                    <div
                      class="locationPickerModalDropdownButton dropdown"
                    >
                      <button
                        aria-expanded="false"
                        class="dropdown-toggle btn btn-outline-secondary"
                        type="button"
                      >
                        Selection
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </body>
        }
      >
        <div
          className="modal-backdrop customBackDropModal show"
          onClick={[Function]}
        />
        <div
          aria-modal={true}
          className="locationPickerModal modal"
          onClick={[Function]}
          onMouseUp={[Function]}
          role="dialog"
          style={
            Object {
              "display": "block",
            }
          }
          tabIndex={-1}
        >
          <ModalDialog
            centered={true}
            className="customWidthModal"
            onMouseDown={[Function]}
            size="lg"
          >
            <div
              className="modal-dialog customWidthModal modal-lg modal-dialog-centered"
              onMouseDown={[Function]}
            >
              <div
                className="modal-content"
              >
                <ModalHeader
                  className="locationPickerModalHeader"
                  closeButton={false}
                  closeLabel="Close"
                >
                  <ForwardRef
                    className="locationPickerModalHeader modal-header"
                    closeButton={false}
                    closeLabel="Close"
                  >
                    <div
                      className="locationPickerModalHeader modal-header"
                    >
                      <div
                        className="locationPickerModalHeaderIcon"
                      >
                        <img
                          alt="location picker icon"
                          src="location-picker.svg"
                        />
                      </div>
                      <h1
                        className="locationPickerModalHeaderTitle"
                      >
                        Please select your location
                      </h1>
                    </div>
                  </ForwardRef>
                </ModalHeader>
                <ModalBody
                  className="locationPickerModalBody"
                >
                  <div
                    className="locationPickerModalBody modal-body"
                  >
                    <DropdownButton
                      className="locationPickerModalDropdownButton"
                      title="Selection"
                      variant="outline-secondary"
                    >
                      <Dropdown
                        align="start"
                        autoClose={true}
                        className="locationPickerModalDropdownButton"
                        navbar={false}
                      >
                        <Dropdown
                          itemSelector=".dropdown-item:not(.disabled):not(:disabled)"
                          onToggle={[Function]}
                          placement="bottom-start"
                        >
                          <div
                            className="locationPickerModalDropdownButton dropdown"
                          >
                            <DropdownToggle
                              variant="outline-secondary"
                            >
                              <Button
                                active={false}
                                aria-expanded={false}
                                className="dropdown-toggle"
                                disabled={false}
                                onClick={[Function]}
                                variant="outline-secondary"
                              >
                                <button
                                  aria-expanded={false}
                                  className="dropdown-toggle btn btn-outline-secondary"
                                  disabled={false}
                                  onClick={[Function]}
                                  type="button"
                                >
                                  Selection
                                </button>
                              </Button>
                            </DropdownToggle>
                            <DropdownMenu
                              flip={true}
                            />
                          </div>
                        </Dropdown>
                      </Dropdown>
                    </DropdownButton>
                  </div>
                </ModalBody>
              </div>
            </div>
          </ModalDialog>
        </div>
      </Portal>
    </Modal>
  </Modal>
</LocationPickerModal>
`;
